main.py

```
@@ -0,0 +1,245 @@
+                raise HTTPException(status_code=429, detail="Rate limit exceeded")
+            rate_limit_store[client_ip] = (last_request_time, request_count + 1)
+        else:
+            rate_limit_store[client_ip] = (current_time, 1)
+    else:
+        rate_limit_store[client_ip] = (current_time, 1)
+    
+    response = await call_next(request)
+    return response
+
+@app.websocket("/ws")  # Annotation for the WebSocket endpoint
+async def websocket_endpoint(websocket: WebSocket, token: str):
```

Ruff (0.8.2)

`websocket_endpoint` is too complex (15 > 10)

(C901)